network/wsNetwork.go

```diff
@@ -1165,36 +1163,129 @@ func (wn *WebsocketNetwork) sendFilterMessage(msg IncomingMessage) {
 
 func (wn *WebsocketNetwork) broadcastThread() {
 	defer wn.wg.Done()
-	var peers []*wsPeer
+
 	slowWritingPeerCheckTicker := time.NewTicker(wn.slowWritingPeerMonitorInterval)
 	defer slowWritingPeerCheckTicker.Stop()
+	peers, lastPeersChangeCounter := wn.peerSnapshot([]*wsPeer{})
+	// updatePeers update the peers list if their peer change counter has changed.
+	updatePeers := func() {
+		if curPeersChangeCounter := atomic.LoadInt32(&wn.peersChangeCounter); curPeersChangeCounter != lastPeersChangeCounter {
+			peers, lastPeersChangeCounter = wn.peerSnapshot(peers[:0])
+			// clear out the unused portion of the peers array to allow the GC to cleanup unused peers.
```

Comment:
Would it make sense to do the GC cleanup inside peerSnapshot() since we're already dealing with the len() and cap() of the passed in value there? Pass in not `peers[:0]` but plain `peers` and inside peerSnapshot() before writing new peer values, nil-out any excess slots if the new peers list is shorter.

Reply:
I think that it's a great idea. I will implement that and update the review.